src/inputfile.rs

```diff
@@ -1,52 +1,159 @@
 use std::fs::File;
-use std::io::{self, BufRead, BufReader};
+use std::io::{self, BufRead, BufReader, ErrorKind};
 
 use content_inspector::{self, ContentType};
 
 use crate::errors::*;
 
 const THEME_PREVIEW_FILE: &[u8] = include_bytes!("../assets/theme_preview.rs");
+const MAXIMUM_LINE_LENGTH: usize = 1024 * 1024;
+
+/// This function is like the read until present in the standard library
```

Comment:
If someone picks this PR up: There is no need to duplicate this function. We can use `take()` instead. See https://github.com/sharkdp/bat/pull/1972/files#r763022008